### PR TITLE
tests: clean up fifos

### PR DESCRIPTION
**Pull request by mr-cal**
- [x] Have you followed the [guidelines for contributing](https://github.com/canonical/snapcraft/blob/main/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox run -m lint`?
- [x] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

Fifos get cached in `.tox/` when running tests and then cause the snapcraft snap to fail to build when snapcraft tries to copy the fifo.

source: @sergio-costas via https://chat.canonical.com/canonical/pl/6jpni1n98ifc3fph9ywmdus4iy

### CODE REVIEW

**Comment by mr-cal** [comment]
> I manually deleted the FIFOs inside the .tox folder, as this patch does, and it solved the problem. I re-run the TOX tests and building the snap failed again. I deleted again the FIFOs, added this patch, re-run the TOX tests, and tried again to build the snap, and this time it worked as expected. So this patch fixes the problem.

Thanks for testing!!

**Review by bepri — APPROVED** [review]
Thanks!

File: tests/legacy/unit/test_elf.py [hunk 1]
```diff
@@ -357,6 +357,7 @@ def test_fifo(self):
 
         elf_files = elf.get_elf_files(self.fake_elf.root_path, {"fifo"})
         self.assertThat(elf_files, Equals(set()))
+        os.unlink(fifo_path)
```
Comment (lengau) [inline tests/legacy/unit/test_elf.py:360]:
```suggestion
        try:
            elf_files = elf.get_elf_files(self.fake_elf.root_path, {"fifo"})
            self.assertThat(elf_files, Equals(set()))
        finally:
            os.unlink(fifo_path)
```

File: tests/unit/elf/test_elf_utils.py [hunk 2]
```diff
@@ -94,6 +94,7 @@ def test_fifo(self, new_dir):
         os.mkfifo(fifo_path)
         elf_files = elf_utils.get_elf_files_from_list(new_dir, {"fifo"})
         assert elf_files == []
+        os.unlink(fifo_path)
```
Comment (lengau) [inline tests/unit/elf/test_elf_utils.py:97]:
```suggestion
        try:
            elf_files = elf_utils.get_elf_files_from_list(new_dir, {"fifo"})
            assert elf_files == []
        finally:
            os.unlink(fifo_path)
```

**Review by lengau — CHANGES_REQUESTED** [review]

File: tests/legacy/unit/test_elf.py (hunk 1 above)
Comment (mr-cal) [reply tests/legacy/unit/test_elf.py:360]: oop thanks!

**Review by lengau — APPROVED** [review]
🚀

**Pull request merged**